stan/math/opencl/prim/bernoulli_cdf.hpp

```diff
@@ -49,13 +49,13 @@ return_type_t<T_prob_cl> bernoulli_cdf(const T_n_cl& n,
                                       theta_val, "in the interval [0, 1]");
   auto theta_bounded_expr = 0.0 <= theta_val && theta_val <= 1.0;
 
-  auto any_n_negative = colwise_max(constant(0, N, 1) + (n < 0));
+  auto any_n_negative = colwise_max(cast<char>(n < 0));
   auto cond = n >= 1;
   auto Pi_uncond = 1.0 - theta_val;
   auto Pi = select(cond, INFTY, Pi_uncond);
   auto P_expr = colwise_prod(select(cond, 1.0, Pi_uncond));
 
-  matrix_cl<double> any_n_negative_cl;
+  matrix_cl<char> any_n_negative_cl;
```

Comment:
Why is this a car and not a bool?

Reply:
OpenCL standard does not specify what is the size of bool, so variables of type bool effectively can not be transferred between host and device (as we need to specify size of the array we are transferring).

Reply:
Can you link me where it says that? I'm only seeing in section [6.6.1](https://www.khronos.org/registry/OpenCL/specs/opencl-1.2.pdf) that bool is expanded to an integer 

> bool: A conditional data type which is either true or false. The value true expands to the integer constant 1 and the value false expands to the integer constant 0.

Reply:
The important thing is what is not in the standard. It never mentions what is the size of bool or what is equivalent host type. Equivalent host types are listed in table 6.1 and for bool it says n/a.

Reply:
Okay I found the relevant docs

https://stackoverflow.com/questions/4441280/does-opencl-support-boolean-variables

> Arguments to __kernel functions in a program cannot be declared with the built-in scalar types bool, half, size_t, ptrdiff_t, intptr_t, and uintptr_t. The size in bytes of these types except half are implementation-defined and in addition can also be different for the OpenCL device and the host processor making it difficult to allocate buffer objects to be passed as arguments to a kernel declared as pointer to these type